scipy/sparse/_coo.py

```diff
@@ -298,15 +299,11 @@ def toarray(self, order=None, out=None):
             M, N = self.shape
             coo_todense(M, N, self.nnz, self.row, self.col, self.data,
                         B.ravel('A'), fortran)
-        else:
-            if fortran:
-                strides = np.append(1, np.cumprod(self.shape[:-1]))
-            else:
-                strides = np.append(np.cumprod(self.shape[1:][::-1])[::-1], 1)
-            coords = np.concatenate(self.coords)
-            coo_todense_nd(strides, self.nnz, self.ndim,
-                           coords, self.data, B.ravel('A'), fortran)
-        # Note: reshape() doesn't copy here, but does return a new array (view).
+
```

Comment:
Just a note here that this removes use of `coo_todense_nd` in favor of the faster `ravel_multi_index/np.add.at`. We should do explore whether to do the same for the 1D case (~6 lines above) by using `coo_todense`... that would avoid including the `coo_todense_nd` function in sparsetools. But I think that could be a separate PR which also explores whether speedups in the code for `coo_todense_nd` are possible. (I recall the speed-up idea being to construct the strides in c++ instead of python.)  Let's look into that separately.